### DISCUSSION TITLE
CSV encode to ByteBuffer?

### DISCUSSION THREAD

**Discussion by javafanboy** [osiegmar/FastCSV#100] [Q&A] [answered]
I would like to CSV Encode records of data to a ByteBuffer without any buffering that when the buffer reaches a specific size it will go through other processing steps like compression, BASE64 encoding and transmission that all work on ByteBuffers. 
I can see that fast CSV integrates with StringWriter but I would for performance reasons like to "leave Strings behind" and work with ByteBuffer instead.
Any thoughts on if this is possible and in that case how? I would not mind subclassing the writer class etc. - just looking for advice on the best performing and simplest way yo do this...

**Comment by osiegmar** [comment, marked as answer]
> I can see that fast CSV integrates with StringWriter

A StringWriter is usually used for demonstration purposes only. The usual use pattern (write to a CSV file) is based on an OutputStreamWriter (see `CsvWriter.CsvWriterBuilder#build(java.nio.file.Path, java.nio.file.OpenOption...)`).

> Any thoughts on if this is possible and in that case how?

You may simply try it by changing the implementation of `CsvWriter.FastBufferedWriter` to use a ByteBuffer/CharBuffer instead of a `char[] buf` and a `WritableByteChannel` instead of a `Writer`.

Then do extensive JMH benchmarks.
